### PR TITLE
Interaction changes for clothing and several reagent containers.

### PR DESCRIPTION
**Pull request by Ghommie**
Just need to give it some nice throughout tests, the most has been done. reviews pretty much needed.

:cl:
tweak: Modified the interaction of several items and reagents with shielded mouth/eyes.
bugfix: Synths can now eat food, no more "monitor heads" messages despite being aesthetically human-like.
bugfix: fixes consumables' temperature adjustment yet again, you shouldn't get cooked to death by pepper spray anymore.
tweak: It's no longer possible to ingest pills, food and drinks through mouth-covering equipment. 
tweak: To ease this, a handful of masks (coifs, clown masks, etc.) are now not as mouth covering as before.
imageadd: ported in a few masks' sprites.
tweak: Non-marine gas masks aren't as vision impairing, but come without the gas/smoke protection. Marine/PMC ones have been made into subtypes to retain their stats. 
tweak: Tweaked coefficients and armor stats on many non-mil clothings.
/:cl:

### CODE REVIEW

**Comment by spookydonut** [comment]
split this up more please so we can merge the stuff thats done and ready sooner.

**Comment by DominikPanic** [comment]
Conflicts.

**Comment by Rohesie** [comment]
This could really use some atomizing.

**Comment by Ghommie** [comment]
I'll see what I can do.

**Comment by Rohesie** [comment]
Boiler gas doesn't work with this PR in.

**Comment by Rohesie** [comment]
Conflicts.

File: code/__HELPERS/game.dm [hunk 1]
```diff
@@ -406,7 +418,33 @@ datum/projectile_data
...
+/proc/mixOneColor(var/list/weight, var/list/color)
+	if (!weight || !color || length(weight)!=length(color))
+		return 0
 
+	var/contents = length(weight)
+	var/i
+
+	//normalize weights
+	var/listsum = 0
+	for(i=1; i<=contents; i++)
```
Comment (Rohesie) [inline code/__HELPERS/game.dm]:
```suggestion
	for(i = 1 to contents)
```

File: code/__HELPERS/game.dm [hunk 2]
```diff
@@ -406,7 +418,33 @@ datum/projectile_data
...
+		listsum += weight[i]
+	for(i=1; i<=contents; i++)
```
Comment (Rohesie) [inline code/__HELPERS/game.dm]:
```suggestion
	for(i = 1 to contents)
```

File: code/__HELPERS/game.dm [hunk 3]
```diff
@@ -406,7 +418,33 @@ datum/projectile_data
...
+		weight[i] /= listsum
+
+	//mix them
+	var/mixedcolor = 0
+	for(i=1; i<=contents; i++)
```
Comment (Rohesie) [inline code/__HELPERS/game.dm]:
```suggestion
	for(i = 1 to contents)
```

File: code/game/gamemodes/colonialmarines/halloween2016.dm [hunk 4]
```diff
@@ -507,7 +507,7 @@
 				H.equip_to_slot_or_del(new /obj/item/clothing/under/marine/veteran/PMC/leader(H), SLOT_W_UNIFORM)
 				H.equip_to_slot_or_del(new /obj/item/clothing/suit/storage/marine/veteran/PMC/leader(H), SLOT_WEAR_SUIT)
 				H.equip_to_slot_or_del(new /obj/item/clothing/head/helmet/marine/veteran/PMC/leader(H), SLOT_HEAD)
-				H.equip_to_slot_or_del(new /obj/item/clothing/mask/gas/PMC/leader(H), SLOT_WEAR_MASK)
+				H.equip_to_slot_or_del(new /obj/item/clothing/mask/gas/marine/PMC/leader(H), SLOT_WEAR_MASK)
```
Comment (Rohesie) [inline code/game/gamemodes/colonialmarines/halloween2016.dm]: Could you nuke this?

File: code/game/objects/effects/effect_system/smoke.dm [hunk 5]
```diff
@@ -7,198 +7,219 @@
...
+		if(S.amount > 0)
+			newsmokes.Add(S)
+		else
+			S.lifetime += rand(-1,1)
+	if(opaque)
+		SetOpacity(TRUE)
+	lifetime += rand(-1,1)
+
+	if(newsmokes.len)
+		spawn(1) //the smoke spreads rapidly but not instantly
```
Comment (Rohesie) [inline code/game/objects/effects/effect_system/smoke.dm]: Can you use a callback here?

File: code/game/objects/items/devices/PDA/PDA.dm [hunk 6]
```diff
@@ -769,16 +769,16 @@ var/global/list/obj/item/device/pda/PDAs = list()
 		empulse(P.loc, 3, 6, 1)
 		message += "Your [P] emits a wave of electromagnetic energy!"
 	if(i>=25 && i<=40) //Smoke
-		var/datum/effect_system/smoke_spread/chem/S = new /datum/effect_system/smoke_spread/chem
+		var/datum/effect_system/smoke_spread/S = new
```
Comment (Rohesie) [inline code/game/objects/items/devices/PDA/PDA.dm]: Just `new` ?

File: code/game/objects/items/storage/briefcase.dm [hunk 7]
```diff
@@ -14,36 +14,28 @@
...
-	//..()
-
+/obj/item/storage/briefcase/attack(mob/living/M, mob/living/user, def_zone)
 	if ((CLUMSY in user.mutations) && prob(50))
 		to_chat(user, "<span class='warning'>The [src] slips out of your hand and hits your head.</span>")
 		user.take_limb_damage(10)
 		user.KnockOut(2)
 		return
 
+	..()
```
Comment (Rohesie) [inline code/game/objects/items/storage/briefcase.dm]:
```suggestion
	. = ..()
```

File: code/game/objects/items/tools/kitchen_tools.dm [hunk 8]
```diff
@@ -191,13 +191,14 @@
...
-/obj/item/tool/kitchen/rollingpin/attack(mob/living/M as mob, mob/living/user as mob)
-	if ((CLUMSY in user.mutations) && prob(50))
+/obj/item/tool/kitchen/rollingpin/attack(mob/living/M, mob/living/user, def_zone)
+	if((CLUMSY in user.mutations) && prob(50))
 		to_chat(user, "<span class='warning'>The [src] slips out of your hand and hits your head.</span>")
 		user.take_limb_damage(10)
 		user.KnockOut(2)
 		return
 
+	..()
```
Comment (Rohesie) [inline code/game/objects/items/tools/kitchen_tools.dm]:
```suggestion
	. = ..()
```

File: code/game/gamemodes/colonialmarines/halloween2016.dm (hunk 4 above)
Comment (Ghommie) [reply code/game/gamemodes/colonialmarines/halloween2016.dm]: As in, removing the commented code?

File: code/game/objects/items/devices/PDA/PDA.dm (hunk 6 above)
Comment (Ghommie) [reply code/game/objects/items/devices/PDA/PDA.dm]: Shamefully copied from the DM help section:

> if Type is not specified and new() is being used in an assignment, the variable type of the left-hand-side will be used as the default type.

So, yea, just the `new`.

File: code/game/gamemodes/colonialmarines/halloween2016.dm (hunk 4 above)
Comment (Rohesie) [reply code/game/gamemodes/colonialmarines/halloween2016.dm]: Well, the whole game mode, I guess, but that may be better in a different PR.

**Comment by DominikPanic** [comment]
Agreed with spooky here, atomize this.

**Comment by spookydonut** [comment]
Atomise the bugfixes and other tweaks into discrete PRs so they can be merged and not held up by this mess.

**Comment by Rohesie** [comment]
Atomize.

**Comment by DominikPanic** [comment]
This contains way too many issues and parts of code that are now in other PRs (even from you), feel free to create new ones that are atomized.

**Pull request closed without merging**